Sources/xcproj/PBXProj+Helpers.swift

```diff
@@ -1,6 +1,18 @@
 import Foundation
 import PathKit
 
+// MARK: - PBXProj Extension (Public)
+
+public extension PBXProj {
+
+    /// Returns project's root group
+    public var rootGroup: PBXGroup {
+        let groupRef = objects.projects[rootObject]!.mainGroup
```

Comment:
I'd make this property optional and remove the force unwraps from this getter

Reply:
I was considering that of course but it will lead to always force unwrapping or checking for nil when using this API, which will make it inconvenient. Is it possible that a project can have no rootObject or group with such reference? If so probably it's a malformed project, so failing should be expected in this case or this should be validated even earlier. As a user I would prefer convenience over safety in this particular case.

Reply:
I also don't like forced unwapping. It can be legitimately be missing when you are constructing project file from scratch. In that case it would be bad behaviour to crash in runtime. Instead this should be either optional variable or a function that throws an error.

Reply:
I would create helper for accessing `rootProject`.
``` swift
	var rootProject: PBXProject? {
		return objects.projects[rootObject]
	}
```
This would make usage of `rootObject` different. Instead you would need to access it through project: `.rootProject?.mainGroup`. But it would be more clear what is optional.

Reply:
> Is it possible that a project can have no rootObject or group with such reference?

There are two usages of xcproj that I can think of right now:
- Opening a project and modify it.
- Create a new project.

In the first case, it's very unlikely that we force unwrap a nil value since the project already has a root group. However, in the second case, it might happen. This is a bad usage sign because the developer should have created the root group before accessing it. 

It depends on whether we want the library to crash at runtime, or execute with unexpected behavior. Thinking about a developer consuming the API, I'd prefer the crash rather than spending time debugging where the unexpected behavior might be coming from.

What do you think @allu22 ?

Reply:
Example where `rootObject` `id` is set and root project does not exist until `generate` is called: 
https://github.com/yonaskolb/XcodeGen/blob/46b1512c7c4dde0621c25e4388d5e02823d3f6ab/Sources/XcodeGenKit/PBXProjGenerator.swift#L33

I think there is no right or wrong approach here. The question is, how defensive you want to be in the API. Either way is fine by me. But in case of unwrapping it here, maybe call `fatalError` with message that hints that you are missing a project object?

Reply:
How about this?

```swift
    public var rootGroup: PBXGroup {
        guard let rootProject = self.rootProject else {
            fatalError("Missing root project")
        }
        guard let rootGroup = objects.groups[rootProject.mainGroup] else {
            fatalError("Root project has no root group")
        }
        return rootGroup
    }
```